lib/assessment.rb

```diff
@@ -20,20 +27,39 @@
 # Name and location should be writeable, but age should be read-only.
 
 ##
-# your code here
+class Person
+  def initialize(name, age, location)
+    @name, @age, @location = name, age, location
+  end
+
+  def set_name(name)
+    @name = name
+  end
+
+  def set_location(location)
+    @location = location
+  end
+
+  attr_reader :age
+
+end
 ##
 
 # Question 4
 # Create a new Person object using parameters "Dave", 32, and "Ohio".
 # Then, set that Person's location to "Somerville"
 
-Answer.dave = nil # replace nil with your code
-Answer.dave.location = nil # replace nil with your code
+Answer.dave = Person.new("Dave",32,"Ohio")
+Answer.dave.location = "Somerville"
 
 # Question 5
 # Create another class called Developer that inherits from Person.
 # Give it a new public method called 'hire_for_job'
 
 ##
-# your code here
+class Developer < Person
+  def hire_for_job
+    puts "you're hired, #{name}!"
```

Comment:
This will work if you're using `attr_reader` or `attr_accessor` on Person (or, alternatively, on Developer), because when you write `name`, it calls the newly generated method 'name'. However, if we hadn't included `attr_...`, this would not have worked; we need to refer to instance variables (such as 'name'), with an '@'; e.g. `@name`.